tests/test_managed_group.rs

```diff
@@ -0,0 +1,341 @@
+        str::from_utf8(&managed_group.client_id()).unwrap(),
+        str::from_utf8(managed_group.member(sender.as_leaf_index()).get_identity()).unwrap(),
+        str::from_utf8(
+            managed_group
+                .member(LeafIndex::from(remove_proposal.removed))
+                .get_identity()
+        )
+        .unwrap(),
+    );
+}
+/// Event listener function for UpdateProposals
+/// `(managed_group: &ManagedGroup, aad: &[u8], sender: &Sender,
+/// update_proposal: &UpdateProposal)`
+fn member_updated(
+    managed_group: &ManagedGroup,
+    _aad: &[u8],
+    sender: &Sender,
+    _update_proposal: &UpdateProposal,
+) {
+    println!(
+        "UpdateProposal received in group {} by {}: {}",
+        str::from_utf8(&managed_group.group_id().as_slice()).unwrap(),
+        str::from_utf8(&managed_group.client_id()).unwrap(),
+        str::from_utf8(managed_group.member(sender.as_leaf_index()).get_identity()).unwrap(),
+    );
+}
+/// Event listener function for application messages
+/// `(managed_group: &ManagedGroup, aad: &[u8], sender: &Sender, message:
+/// &[u8])`
+fn app_message_received(
+    managed_group: &ManagedGroup,
+    _aad: &[u8],
+    sender: &Sender,
+    message: &[u8],
+) {
+    println!(
+        "Message received in group {} by {} from {}: {}",
+        str::from_utf8(&managed_group.group_id().as_slice()).unwrap(),
+        str::from_utf8(&managed_group.client_id()).unwrap(),
+        str::from_utf8(managed_group.member(sender.as_leaf_index()).get_identity()).unwrap(),
+        str::from_utf8(message).unwrap()
+    );
+}
+/// Event listener function for invalid messages
+/// `(managed_group: &ManagedGroup, aad_option: Option<&[u8]>, sender_option:
+/// Option<&Sender>, error: InvalidMessageError)`
+fn invalid_message_received(
+    managed_group: &ManagedGroup,
+    aad_option: Option<&[u8]>,
+    sender_option: Option<&Sender>,
+    error: InvalidMessageError,
+) {
+    println!(
+        "Invalid message received in group {} by {}, error: {}",
+        str::from_utf8(&managed_group.group_id().as_slice()).unwrap(),
+        str::from_utf8(&managed_group.client_id()).unwrap(),
+        error,
+    );
+    if let Some(aad) = aad_option {
+        println!(" >>> AAD: {:?}", aad);
+    }
+    if let Some(sender) = sender_option {
+        println!(
+            " >>> Sender: {:?}",
+            managed_group.member(sender.as_leaf_index())
+        );
+    }
+}
+
+/// This test simulates various group operations like Add, Update, Remove in a
+/// small group
+///  - Alice creates a group
+///  - Alice adds Bob
+///  - Alice sends a message to Bob
+///  - Bob updates and commits
+///  - Alice updates and commits
+///  - Bob adds Charlie
+///  - Charlie sends a message to the group
+///  - Charlie updates and commits
+///  - Charlie removes Bob
+#[test]
+#[should_panic]
+fn managed_group_operations() {
+    for ciphersuite in Config::supported_ciphersuites() {
```

Comment:
Ok for a first version. But we should start making these individual tests, i.e. put the test in a macro and call it for all ciphersuites. Otherwise we'll get test failures that are impossible to debug.